parcels/xgrid.py

```diff
@@ -314,6 +316,33 @@ def _fpoint_info(self):
 
         return axis_position_mapping
 
+    def get_spatial_hash(
+        self,
+        global_grid=False,
+        reconstruct=False,
+    ):
```

Comment:
Out of interest: Does uxarray internals use spatial hashing by default? (i.e., there's no need for us to work with a uxarray spatial hash object, operations with uxarray just use it under the hood). If so, then I think its good to keep this method on the XGrid only